src/components/color_picker/color_picker.ts

```diff
@@ -246,50 +319,99 @@ export class ColorPicker extends Component<ColorPickerProps, SpreadsheetChildEnv
     this.props.onColorPicked("");
   }
 
-  setCustomColor(ev: Event) {
-    if (!isColorValid(this.state.currentColor)) {
+  AddCustomColor(ev: Event) {
+    if (!isHSLAValid(this.state.currentHslaColor) || !isColorValid(this.state.customHexColor)) {
       ev.stopPropagation();
       this.state.isCurrentColorInvalid = true;
       return;
     }
-    const color = toHex(this.state.currentColor);
+    ev.stopPropagation();
```

Comment:
I'm not sure it's really needed, but the `ev.stopPropagation` can be put before the if to avoid writing it twice in both cases